### DISCUSSION TITLE
Nextcloud AIO Eats All My Memory

### DISCUSSION THREAD

**Discussion by Lwfrancisco** [nextcloud/all-in-one#944] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Update Nextcloud AIO to latest
2. Unsure what causes it exactly, but I've now caused it by syncing via the Windows Desktop app and again while deleting 3 files in the webui.
3. Notice RAM usage spikes to maximum on the system.

### Expected behavior 
Delete and sync files.

### Actual behavior 
System freezes or slows down terribly as RAM usage spikes. This LXC container ONLY has nextcloud-aio on it.

### Host OS 
Proxmox LXC - Debian

#### Nextcloud AIO version 
Latest

#### Current channel 
Stable

#### Other valuable info 
The host system is not by any means overloaded. Neither is the Container. Started on 4GB RAM, but encountered this issue. Upgraded container to 6GB RAM and again encountered this issue shortly after.

This is a low volume server, with only a handful of users and in all likelihood only 1 concurrent user throughout.

The first time, when this was encountered during a "sync", I was able to open top and see that php and postgres had spawned a ton of processes. I don't know if this is particularly normal but there seemed to be quite a few processes. The second time, now with 6GB of RAM, the system just totally froze up and could not run top after half an hour.

Please let me know what logs I need to provide!

Screenshot of both instances (also shows approximately how long it took after the system was again up and running for this to happen once more).
![ProxmoxNextcloudAIOCapture](https://user-images.githubusercontent.com/8664661/178095372-aba4651c-b50a-4fa2-bf39-35d7a3168b1b.PNG)

**Comment by szaimen** [comment]
Hm... Are you using the external storage app in your Nextcloud?

Also how many desktop clients are set up to sync files? And does the sync finish successfully (or does it never end and always starts over again)?

Is this spike in ram usage always happening in relation to the desktop client?

**Comment by Lwfrancisco** [comment]
Nope, no external storage app. Also, likely only one desktop sync is set up. Mayhaps 2 phone syncs.

The sync will most often finish. I've only encountered this weird behavior a few times and no consistent usage pattern. Is there a log I can grab that would help clarify this?

**Comment by szaimen** [comment]
I fear I don't have any clue what could trigger this then....

**Comment by szaimen** [comment]
I'll move this to discussions for now as I cannot reproduce this...

**Comment by Calambro** [comment]
In my case I installed Nextcloud AIO three days ago and have been experiencing similar spikes in CPU usage.
- latest
- stable
- Ubuntu 22.04 LTS server
- 2 GB RAM

My Nextcloud froze not long after I created the instance and came back to normal after about an hour. This happened again around 15 hours later. It happened yet again 36 hours later and lasted for about 8.5 hours. This last time was moslty while I was sleeping, so I was not doing anything with it then.  


<img width="446" alt="Screenshot 2022-07-16 002140" src="https://user-images.githubusercontent.com/109383359/179417505-56450370-0e47-4ab2-ad75-a6a62827ba21.png">
<img width="883" alt="Screenshot 2022-07-17 102939" src="https://user-images.githubusercontent.com/109383359/179417507-8ac0b0c6-6a3e-42cb-8132-3896360f53be.png">
.

**Comment by szaimen** [comment, marked as answer]
Is this still happening with v2.0.0 Beta? See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by Lwfrancisco** [reply]
To be honest, this hasn't happened in at least a month. I'm guessing one of the container updates awhile back caused this odd behavior and has since been mitigated. Recently, I've also had more users than usual on my instance (a handful total) as most of us took a vacation together and used it to share photos and media around.

**Reply by szaimen** [reply]
thanks for the feedback! Marking then as resolved for now.
